### PR TITLE
test: airgapped deployments

### PR DESCRIPTION
**Pull request by evan-onyx**
## Description

new tests to check whether airgapped deployments work. the unit test is hitting a specific issue we saw, while the two new airgap env tests are targeting a) the basic "happy path" case of "does deployment work when u just deny egress" as well as the specific edge case of "connectivity exists but you get tls errors".

The deployment checks are integrated into the existing Playwright workflow behind a dedicated startup-path filter. They reuse its prebuilt ARM64 backend and model-server images instead of launching duplicate image-build jobs, while retaining the isolated no-egress and untrusted-TLS health scenarios.

## How Has This Been Tested?

Confirmed that the airgap tests and the new unit test fail in Ci in the current repo state.

## Additional Options

- [x] [Optional] Please cherry-pick this PR to the latest release version.
- [x] [Optional] Override Linear Check

---
## Summary by cubic
Adds CI and tests for airgapped deployments (no‑egress and untrusted TLS). Upgrades the model stack (`transformers` 5.14.1, `huggingface-hub` 1.26.0), bundles the default embedding model, and loads it from local files with remote code disabled to keep embeddings finite and fully offline.

- **New Features**
  - New `airgap-health` job reuses prebuilt images, runs no‑egress and self‑signed TLS Compose stacks, waits on health, validates embeddings in‑container, uploads logs, and gates PRs on relevant changes.
  - New Compose overlays: `docker-compose.airgap-test.yml` sets `HF_ENDPOINT`, CI‑only `USER_AUTH_SECRET`, internal network, and healthchecks; `docker-compose.airgap-tls-test.yml` adds a `tls_failure` service at https://tls_failure:4443.

- **Bug Fixes**
  - Model server loads the default embedder from local files only with remote code disabled; adds tests to ensure cached tokenizers load when Hugging Face TLS verification fails across `huggingface-hub` versions.
  - Airgap CI stability: added healthchecks, `docker compose --wait` timeouts, and explicit `api_server` health probing.

<sup>Written for commit 0e033220fb5bf06e3caebb58faaee26d4ae82007. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/onyx-dot-app/onyx/pull/13733?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: .github/workflows/pr-airgap-deployment-tests.yml
Comment (justin-tahara) [inline .github/workflows/pr-airgap-deployment-tests.yml:1]: I have a feeling that we don't need to build out all of the images again but I could be wrong and we would need to move things around to get that to work. Worth double checking though. I feel like Jamison has spent a lot of time making improvements for things like this

**Review by justin-tahara — APPROVED** [review]

File: backend/model_server/encoders.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def _prewarm_rope(st_model: "SentenceTransformer", target_len: int) -> None:
         logger.notice("Loading %s", model_name)
         model = SentenceTransformer(
             model_name_or_path=model_name,
-            trust_remote_code=True,
+            trust_remote_code=False,
```
[the remark this answers is not in the document]
Comment (evan-onyx) [reply backend/model_server/encoders.py:62]: we're on 5.14.1 now
